### DISCUSSION TITLE
Helix: a way to select random theme

### DISCUSSION THREAD

**Discussion by valerii15298** [helix-editor/helix#6892] [Q&A] [answered]
Is there any way to use random theme when opening helix every time?

**Comment by kirawi** [comment, marked as answer]
No, and it won't be a feature in Helix. You have a few options instead:
- (a) Write a script that will use a random theme and write it to `my_theme.toml`. In your `config.toml`, you can set it to use `my_theme` by default.
- (b) Write a script that will modify the `config.toml` to use a random theme.

You can create an alias for running the script and then opening Helix.

**Reply by elwinar** [reply]
I've done just that before even checking if it was supported. Here is my script for anyone who's lazy enough :

```
#! /usr/bin/env bash

theme=$(find "/usr/lib/helix/runtime/themes" -name "*.toml" | xargs basename -s ".toml" | shuf -n 1)
echo "inherits=\"$theme\"" > ~/.config/helix/themes/random.toml
```

I've set it to run when my DE start, so it changes at every boot. Adding a dumb `helix $@` at the end would allow one to use it as an alias. Note that I'm limiting myself to the stock themes, likewise one could use a special repository with symlinks or something like that to get stock themes, user-defined, etc.
